tests/controller/test_modelrun.py

```diff
@@ -171,3 +187,92 @@ def test_call_simulate(self, mock_store, mock_model_run):
         runner = ModelRunner()
         runner.solve_model(mock_model_run, mock_store)
         assert mock_model_run.sos_model.simulate.call_count == 2
+
+class TestModelRunnerJobGraphs():
+    """Check if the ModelRunner glues jobgraphs generated by the SosModel 
+    correctly together
+    """
+    def test_jobgraph_linear(self, mock_store, mock_model_run):
+        """
+        0   (before_model_run_model_a)
+        |
+        0   (simulate_model_b)
+        """
+        mock_model_run.sos_model.before_model_run = Mock(return_value=nx.DiGraph({
+            'before_model_run_model_a': {}
+        }))
+        mock_model_run.sos_model.simulate = Mock(return_value=nx.DiGraph({
+            'simulate_0_2010_model_a': {}
+        }))
+
+        runner = ModelRunner()
+        job_graph = runner.solve_model(mock_model_run, mock_store)
+
+        assert list(job_graph.predecessors('before_model_run_model_a')) == []
+        assert list(job_graph.successors('before_model_run_model_a')) == ['simulate_0_2010_model_a']
+
+        assert list(job_graph.predecessors('simulate_0_2010_model_a')) == ['before_model_run_model_a']
+        assert list(job_graph.successors('simulate_0_2010_model_a')) == []
+
+    def test_jobgraph_convergent(self, mock_store, mock_model_run):
+        """
+        0   0  (before_model_run_model_a, before_model_run_model_b)
+         \ /
+          0    (simulate_model_a)
+          |
+          0    (simulate_model_b)
+        """
+        mock_model_run.sos_model.before_model_run = Mock(return_value=nx.DiGraph({
+            'before_model_run_model_a': {},
+            'before_model_run_model_b': {}
+        }))
+        mock_model_run.sos_model.simulate = Mock(return_value=nx.DiGraph({
+            'simulate_0_2010_model_a': {
+                'simulate_0_2010_model_b': {}
+            }
+        }))
+
+        runner = ModelRunner()
+        job_graph = runner.solve_model(mock_model_run, mock_store)
+
+        assert list(job_graph.predecessors('before_model_run_model_a')) == []
+        assert list(job_graph.successors('before_model_run_model_a')) == ['simulate_0_2010_model_a']
+
+        assert list(job_graph.predecessors('before_model_run_model_b')) == []
+        assert list(job_graph.successors('before_model_run_model_b')) == ['simulate_0_2010_model_a']
+
+        assert sorted(list(job_graph.predecessors('simulate_0_2010_model_a'))) == sorted(['before_model_run_model_a', 'before_model_run_model_b'])
+        assert list(job_graph.successors('simulate_0_2010_model_a')) == ['simulate_0_2010_model_b']
+
+        assert list(job_graph.predecessors('simulate_0_2010_model_b')) == ['simulate_0_2010_model_a']
+        assert list(job_graph.successors('simulate_0_2010_model_b')) == []
+
+    def test_jobgraph_cross(self, mock_store, mock_model_run):
+        """
+        0   0  (before_model_run_model_a, before_model_run_model_b)
+        | X |
+        0   0  (simulate_model_a, simulate_model_b)
```

Comment:
Similarly, here - should each before/simulate pair be independent if there are no other dependencies?
```
a[before]   b[before]
|           |
v           v
a[sim]      b[sim]
```